x-pack/legacy/plugins/snapshot_restore/public/app/sections/home/restore_list/restore_list.tsx

```diff
@@ -207,7 +208,7 @@ export const RestoreList: React.FunctionComponent = () => {
   }
 
   return (
-    <WithPrivileges privileges="index.*">
+    <WithPrivileges privileges={APP_RESTORE_INDEX_PRIVILEGES.map(name => `index.${name}`)}>
```

Comment:
Why are we prepending `index.` here?

Reply:
Because privileges are namespaced by "section". This allows requiring:

* all privileges of a specific section: `"index.*"`
* a specific privilege `"index.some_specific_index_privilege"`

It also prevents privilege naming collision.

Reply:
What is a "section"? Is it just an arbitrary identifier or are we creating formal distinctions between index privileges and cluster privileges?

If it's the latter, then I suggest changing this interface to make this clear and so can we avoid special strings which I find obscure intention.

```js
<WithPrivileges indexPrivileges={APP_RESTORE_INDEX_PRIVILEGES} clusterPrivileges={EXAMPLE_REQUIRED_CLUSTER_PRIVILEGES}>
```

Reply:
A "section" is a "namespace". It allows us to group privileges. Your suggestion would not re-usable as there is an infinite number of groups.